src/Events.php

```diff
@@ -156,6 +157,14 @@ protected function possibly_add_notes() {
 		UpdateStoreDetails::possibly_add_note();
 	}
 
+	/**
+	 * Deletes notes that should be deleted.
+	 */
+	protected function possibly_delete_notes() {
+		NavigationNudge::delete_if_not_applicable();
+		NavigationFeedback::delete_if_not_applicable();
```

Comment:
One more I completely forgot about that could benefit from this!  `NavigationFeedbackFollowUp`

Reply:
I'm creating an issue for `NavigationFeedbackFollowUp`.